test/abbrtypes.ml

```diff
@@ -0,0 +1,73 @@
+(* Slacko's public interface doesn't let us easily construct Slacko.user and
```

Comment:
The tests need to build expected values to compare the actual values to. I was initially just using `channel_of_string` and `user_of_string`, but both of those explode on empty strings and there are places in the API the return empty strings for `user` fields.

This `abbrtypes` thing definitely needs to be replaced with something better -- it's very much a temporary hack to let me move on with the more useful test code without having to solve this problem properly first. :-)